pkg/virtctl/update/machine-type/convert-machine-type/convert_machine_type_test.go

```diff
@@ -0,0 +1,173 @@
+	var vmiFeeder *testutils.VirtualMachineFeeder
+	var controller *JobController
+	var vm *virtv1.VirtualMachine
+	var vmi *virtv1.VirtualMachineInstance
+	var err error
+
+	initController := func() {
+		controller, err = NewJobController(vmiInformer, virtClient)
+		Expect(err).ToNot(HaveOccurred())
+
+		mockQueue = testutils.NewMockWorkQueue(controller.Queue)
+		controller.Queue = mockQueue
+		vmiFeeder = testutils.NewVirtualMachineFeeder(mockQueue, vmiSource)
+	}
+
+	shouldExpectGetVM := func(vm *virtv1.VirtualMachine) {
+		vmInterface.EXPECT().Get(context.Background(), vm.Name, &metav1.GetOptions{}).Return(vm, nil).Times(1)
+
+		kubeClient.Fake.PrependReactor("get", "virtualmachines", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+			get, ok := action.(testing.GetAction)
+			Expect(ok).To(BeTrue())
+			Expect(get.GetNamespace()).To(Equal(metav1.NamespaceDefault))
+			Expect(get.GetName()).To(Equal(vm.Name))
+			return true, vm, nil
+		})
+	}
+
+	shouldExpectVMICreation := func(vmi *virtv1.VirtualMachineInstance) {
+		kubeClient.Fake.PrependReactor("create", "virtualmachineinstances", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+			create, ok := action.(testing.CreateAction)
+			Expect(ok).To(BeTrue())
+			Expect(create.GetObject().(*virtv1.VirtualMachineInstance).Name).To(Equal(vmi.Name))
+
+			return true, create.GetObject(), nil
+		})
+	}
+
+	shouldExpectVMIDeletion := func(vmi *virtv1.VirtualMachineInstance) {
+		kubeClient.Fake.PrependReactor("delete", "virtualmachineinstances", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+			delete, ok := action.(testing.DeleteAction)
+			Expect(ok).To(BeTrue())
+			Expect(delete.GetName()).To(Equal(vmi.Name))
+
+			return true, nil, nil
+		})
+	}
+
+	shouldExpectRemoveLabel := func(vm *virtv1.VirtualMachine) {
+		patchData := `[{"op": "remove", "path": "/metadata/labels/restart-vm-required"}]`
+
+		vmInterface.EXPECT().Patch(context.Background(), vm.Name, types.JSONPatchType, []byte(patchData), &metav1.PatchOptions{}).Times(1)
+
+		kubeClient.Fake.PrependReactor("patch", "virtualmachines", func(action testing.Action) (handled bool, obj runtime.Object, err error) {
+			patch, ok := action.(testing.PatchAction)
+			Expect(ok).To(BeTrue())
+			Expect(patch.GetPatch()).To(Equal([]byte(patchData)))
+			Expect(patch.GetPatchType()).To(Equal(types.MergePatchType))
+			Expect(vm.Labels).ToNot(HaveKey("restart-vm-required"), "should remove `restart-vm-required` label")
+			return true, vm, nil
+		})
+	}
+
+	Describe("When VMI is deleted", func() {
+
+		BeforeEach(func() {
+			ctrl = gomock.NewController(GinkgoT())
+			virtClient = kubecli.NewMockKubevirtClient(ctrl)
+			vmInterface = kubecli.NewMockVirtualMachineInterface(ctrl)
+			vmiInterface = kubecli.NewMockVirtualMachineInstanceInterface(ctrl)
+
+			vmiInformer, vmiSource = testutils.NewFakeInformerFor(&virtv1.VirtualMachineInstance{})
```

Comment:
This informer needs to be injected to the controller. IOW, the controller does not use this informer.